spec/v3/services/repository/update_spec.rb

```diff
@@ -0,0 +1,73 @@
+describe Travis::API::V3::Services::Repository::Update, set_app: true do
+  let(:repo) { Travis::API::V3::Models::Repository.where(owner_name: 'svenfuchs', name: 'minimal').first }
+
+  before { Travis.config.applications = { sync: { full_access: true, token: '12345' } } }
+  after  { Travis.config.applications = {} }
+
+  describe "not authenticated" do
+    before  { patch("/v3/repo/#{repo.id}?app.name=sync", com_id: 1) }
```

Comment:
Why is the param needed if the header is sent?

Reply:
I think that's okay – I just don't get why both the query param and the header are needed? Isn't the header with `app 12345` enough?

Reply:
i suppose it would be now, if we want to depend on the `app` part in the header value ... which is the bit i wasn't sure about